### DISCUSSION TITLE
Does anyone know why the components naming convention changed?

### DISCUSSION THREAD

**Discussion by mxswat** [tailwindlabs/headlessui#3379] [General]
I just learned about the v2 version, and I appreciated the `Component.SubComponent` syntax, and I would like to know why this changed.

I'm interested in knowing this because I used this naming convention even for my own components since I started using headless.

**Comment by mxswat** [comment]
Bump

**Comment by mxswat** [comment]
Bump 2

**Comment by hmaesta** [comment]
> **Deprecated component dot-notation**
> The previous component dot-notation has been deprecated in favor of using explicit imports for each individual component. This is to improve compatibility with RSC (React Server Components) and tree-shaking.

Source: [@headlessui/react@v2.0.0](https://github.com/tailwindlabs/headlessui/releases/tag/%40headlessui%2Freact%40v2.0.0#user-content-upgrading-from-v1:~:text=Deprecated%20component%20dot%2Dnotation)

**Reply by mxswat** [reply]
Thank you, very helpful
